payments/db/test_kvdb.go

```diff
@@ -8,7 +8,24 @@ import (
 )
 
 // NewTestDB is a helper function that creates an BBolt database for testing.
-func NewTestDB(t *testing.T, opts ...OptionModifier) *KVPaymentsDB {
+func NewTestDB(t *testing.T, opts ...OptionModifier) PaymentDB {
```

Comment:
so this one is expected to later change to depend on build tag right? if so, can we add a comment to say this?